module/window_swallowing.lua

```diff
@@ -60,12 +77,12 @@ local function manage_clientspawn(c)
         return
     end
 
-    -- io.popen is normally discouraged. Should probably be changed
-    -- returns "init(1)---ancestorA(pidA)---ancestorB(pidB)...---process(pid)"
-    local handle = io.popen("pstree -A -p -s " .. tostring(c.pid))
-    local parent_pid = handle:read("*a")
-    handle:close()
-
+    get_parent_pid(c.pid, function(err, ppid)
+        if err then
+            error(err)
```

Comment:
What's `error`? I think it's not defined

Reply:
good catch, i think that `error()` was a function that returned/printed/logged the error code on the async snippet i copied the async function and callback from, will remove it.